src/transformers/pipelines/fill_mask.py

```diff
@@ -115,25 +115,53 @@ def __call__(self, *args, targets=None, top_k: Optional[int] = None, **kwargs):
         inputs = self._parse_and_tokenize(*args, **kwargs)
         outputs = self._forward(inputs, return_tensors=True)
 
+        # top_k must be defined
+        if top_k is None:
+            top_k = self.top_k
+
         results = []
         batch_size = outputs.shape[0] if self.framework == "tf" else outputs.size(0)
 
         if targets is not None:
-            if len(targets) == 0 or len(targets[0]) == 0:
-                raise ValueError("At least one target must be provided when passed.")
             if isinstance(targets, str):
                 targets = [targets]
 
-            targets_proc = []
+            vocab = self.tokenizer.vocab
+            target_ids = []
             for target in targets:
-                target_enc = self.tokenizer.tokenize(target)
-                if len(target_enc) > 1 or target_enc[0] == self.tokenizer.unk_token:
+                id_ = vocab.get(target, None)
+                if id_ is None:
+                    input_ids = self.tokenizer(
+                        target,
+                        add_special_tokens=False,
+                        return_attention_mask=False,
+                        return_token_type_ids=False,
+                        max_length=1,
+                        truncation=True,
+                    )["input_ids"]
+                    if len(input_ids) == 0:
+                        logger.warning(
+                            f"The specified target token `{target}` does not exist in the model vocabulary. "
+                            f"We cannot replace it with anything meaningful, irngoring it"
```

Comment:
Wouldn't this return the unknown token if it doesn't exist in the tokenizer vocab?

Reply:
For example:

```py
>>> from transformers import BertTokenizerFast
>>> tokenizer = BertTokenizerFast.from_pretrained("bert-base-cased")
>>> tokenizer("珞", add_special_tokens=False, max_length=1, truncation=True)["input_ids"]
[100]
```
with `100` being the unknown token

Reply:
Not necessarily for the empty string "". (I guess single space could give the same if stripping is part of the tokenization preprocessing step, I am not sure)